torch/lib/c10d/logger.cpp

```diff
@@ -191,6 +198,13 @@ void Logger::set_runtime_stats_and_log() {
     return;
   }
   num_iterations_stats_recorded_++;
+
+  // Check if comm. hook has not been set, if so, set it as builtin allreduce.
+  if (!comm_hook_set_) {
+    ddp_logging_data_->comm_hook = kDefaultDDPComm;
```

Comment:
If no hook is represented by an empty string, this added if statement can be removed.